### PR TITLE
Integrate forbiddenapis + fix various forbiddenapi usage.

### PR DESCRIPTION
**Pull request by johnou**
Motivation:

Checks Java byte code against a list of "forbidden" API signatures.

Modification:

Integrate forbiddenapis plugin and fix usages.

Result:

Future proof code.

### CODE REVIEW

**Comment by johnou** [comment]
@anistor let me know if you would prefer that I split this into multiple pull requests.

**Comment by johnou** [comment]
@anistor this one is ready to go too, if I work out an elegant solution to IPROTO-88 I will uncomment the custom forbidden API..

File: core/src/main/java/org/infinispan/protostream/annotations/impl/types/ReflectionClassFactory.java [hunk 1]
```diff
@@ -109,9 +111,9 @@ public XClass fromTypeMirror(TypeMirror typeMirror) {
                   try {
                      f.setAccessible(true);
                      e = (Enum<?>) f.get(clazz);
-                  } catch (IllegalAccessException iae) {
+                  } catch (IllegalArgumentException | IllegalAccessException ex) {
```
Comment (anistor) [inline core/src/main/java/org/infinispan/protostream/annotations/impl/types/ReflectionClassFactory.java]: I'd rather not handle IllegalArgumentException or any other exceptions here, just let it explode. But IllegalAccessException is not a runtime exception.

**Comment by anistor** [comment]
What is the purpose of the locale related fixes @johnou ? Do we have any failures?

File: core/src/main/java/org/infinispan/protostream/annotations/impl/types/ReflectionClassFactory.java (hunk 1 above)
Comment (johnou) [reply core/src/main/java/org/infinispan/protostream/annotations/impl/types/ReflectionClassFactory.java]: Got it, I'll roll it back. Point was to throw an exception with useful information.

**Comment by johnou** [comment]
@anistor no failures, could technically suppress the check for those if you prefer since there should be no problems with english only messages - let me know what you prefer.

File: core/src/main/java/org/infinispan/protostream/annotations/impl/types/ReflectionClassFactory.java (hunk 1 above)
Comment (johnou) [reply core/src/main/java/org/infinispan/protostream/annotations/impl/types/ReflectionClassFactory.java]: reverted.

**Review by akkarapon26 — APPROVED** [review]

**Comment by johnou** [comment]
let's close this.

**Pull request closed without merging**